scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/job/termination/handlers/TerminateIfTaskHandler.java

```diff
@@ -0,0 +1,190 @@
+		InternalTask targetElse = targets[1];
+		InternalTask targetJoin = targets[2];
+
+		logger.info("Control Flow Action IF: " + targetIf.getId() + " join: "
+				+ ((targetJoin == null) ? "null" : targetJoin.getId()));
+
+		// these 2 tasks delimit the Task Block formed by the IF branch
+		InternalTask branchStart = targetIf;
+		InternalTask branchEnd = null;
+
+		String match = targetIf.getMatchingBlock();
+		if (match != null) {
+			for (InternalTask t : internalJob.getIHMTasks().values()) {
+				if (match.equals(t.getName())
+						&& !(t.getStatus().equals(TaskStatus.FINISHED) || t.getStatus().equals(TaskStatus.SKIPPED))) {
+					branchEnd = t;
+				}
+			}
+		}
+		// no matching block: there is no block, the branch is a single task
+		if (branchEnd == null) {
+			branchEnd = targetIf;
+		}
+
+		// plug the branch
+		branchStart.addDependence(initiator);
+		changesInfo.taskUpdated(branchStart);
+		if (targetJoin != null) {
+			targetJoin.addDependence(branchEnd);
+			changesInfo.taskUpdated(targetJoin);
+		}
+
+		// the other branch will not be executed
+		// first, find the concerned tasks
+		List<InternalTask> elseTasks = new ArrayList<>();
+		// elseTasks.add(targetElse);
+		for (InternalTask t : internalJob.getIHMTasks().values()) {
+			if (t.dependsOn(targetElse)) {
+				elseTasks.add(t);
+			}
+		}
+
+		// even though the targetElse is not going to be executed, a
+		// dependency on initiator still makes sense and would help
+		// reconstruct the job graph on the client
+		targetElse.addDependence(initiator);
+		changesInfo.taskUpdated(targetElse);
+
+		for (InternalTask it : elseTasks) {
+			it.setFinishedTime(descriptor.getFinishedTime() + 1);
+			it.setStatus(TaskStatus.SKIPPED);
+			it.setExecutionDuration(0);
+
+			changesInfo.taskSkipped(it);
+			internalJob.setNumberOfPendingTasks(internalJob.getNumberOfPendingTasks() - 1);
+			internalJob.setNumberOfFinishedTasks(internalJob.getNumberOfFinishedTasks() + 1);
+			logger.info("Task " + it.getId() + " will not be executed");
+		}
+
+		// plug the branch in the descriptor
+		TaskId joinId = null;
+		if (targetJoin != null) {
+			joinId = targetJoin.getId();
+		}
+		internalJob.getJobDescriptor().doIf(initiator.getId(), branchStart.getId(), branchEnd.getId(), joinId,
+				targetElse.getId(), elseTasks);
+
+		((JobInfoImpl) internalJob.getJobInfo()).setTasksChanges(changesInfo, internalJob);
+		// notify frontend that tasks were modified
+		if (frontend != null) {
+			frontend.jobStateUpdated(internalJob.getOwner(),
+					new NotificationData<>(SchedulerEvent.TASK_SKIPPED, internalJob.getJobInfo()));
+		}
+		((JobInfoImpl) internalJob.getJobInfo()).clearTasksChanges();
+
+		// no jump is performed ; now that the tasks have been plugged
+		// the flow can continue its normal operation
+		internalJob.getJobDescriptor().terminate(taskId);
+
+		return true;
+	}
+
+	private InternalTask[] searchIfElseJoinTasks(FlowAction action, InternalTask initiator) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "searchIfElseJoinTasks" is 27 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)